src/bindgen/ir/ty.rs

```diff
@@ -482,8 +482,23 @@ impl Type {
                     }
                     Type::Primitive(prim)
                 } else {
-                    Type::Path(generic_path)
+                    // Check for qself in syn::TypePath
+                    if let Some(qself) = &path.qself{
+                        let qself_type = Type::load(&qself.ty)?;
+                        match qself_type {
+                            Some(ty) => {
+                                generic_path.set_qself(ty);
+                                Type::Path(generic_path)
+                            },
+                            _ => {Type::Path(generic_path)}
+                        }
+                    }
```

Comment:
can we put this into `GenericPath::load`?

Reply:
Reason for this was that `GenericPath::load` takes a `&syn::Path` as its argument. 

The `syn::TypePath` structure is like this: 
pub struct TypePath {
    pub qself: Option<QSelf>,
    pub path: Path,
}

and `GenericLoad::Path` only takes the `syn::TypePath::Path` part. To include the `syn::TypePath::qself` part, I would have to modify the function signature to either take a `TypePath` or to add an additional `Option<Qself>` argument. Should I go ahead and do this? My initial thought was that this might break compatibility with some existing code.

Reply:
> To include the syn::TypePath::qself part, I would have to modify the function signature to either take a `TypePath`

yes, modify the function signature to take `TypePath`

Reply:
Moved to the function, modified `GenericPath::load()` to take an extra argument `&Option<QSelf>`, this is done instead of changing the argument to a singular `syn::TypePath` since this function is actually used in one other place in code where the type to convert is not a `syn::TypePath` (An example is `syn::ExprPath`)

Reply:
fine